File: 398.py
from random import randint
from typing import List


class Solution:

    def __init__(self, nums: List[int]):
        self.nums = nums

    def pick(self, target: int) -> int:
        cnt = 0
        ans = 0
        for i, num in enumerate(self.nums):
            if num == target:
                if randint(0, cnt) == 0:
                    ans = i
                cnt += 1
        return ans


# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.pick(target)

File: 398.go
package main

import "math/rand"

type Solution struct {
	nums []int
}

func Constructor(nums []int) Solution {
	return Solution{
		nums,
	}
}

func (this *Solution) Pick(target int) int {
	var ans int
	cnt := 0
	for i, num := range this.nums {
		if num == target {
			cnt++
			if rand.Intn(cnt) == 0 {
				ans = i
			}
		}
	}
	return ans
}

/**
 * Your Solution object will be instantiated and called as such:
 * obj := Constructor(nums);
 * param_1 := obj.Pick(target);
 */
